bin/menmos-fuse/src/omnifs/cached_client.rs

```diff
@@ -66,40 +69,24 @@ impl CachedClient {
     }
 
     pub async fn create_empty(&self, meta: Meta) -> Result<String> {
-        {
-            let mut cache_guard = self.query_cache.lock().await;
-            let cache = &mut *cache_guard;
-            cache.clear();
-        }
+        self.query_cache.clear().await;
         Ok(self.client.create_empty(meta).await?)
     }
 
     pub async fn write(&self, blob_id: &str, offset: u64, buffer: Bytes) -> Result<()> {
-        {
-            let mut cache_guard = self.meta_cache.lock().await;
-            let cache = &mut *cache_guard;
-            cache.invalidate(blob_id);
-        }
+        self.meta_cache.clear().await;
         Ok(self.client.write(blob_id, offset, buffer).await?)
     }
 
     pub async fn update_meta(&self, blob_id: &str, meta: Meta) -> Result<()> {
-        {
-            let mut cache_guard = self.query_cache.lock().await;
-            let cache = &mut *cache_guard;
-            cache.clear();
-        }
+        self.query_cache.clear().await;
```

Comment:
Can you briefly explain why you clear the entire cache on update/write and not simply "update" it?

Reply:
Updating the cache manually would work but would be non-trivial to implement. The update meta call is used (among other things) to move a blob from one parent to another, so _all_ the queries containing this blob ID as a parent can potentially see their result set change. For now it's safer to simply dump the query cache to avoid spooky behavior (also it's not that bad to do since queries are so fast 🔥 )

Reply:
Right!